<a href="https://colab.research.google.com/github/shabana1408/Project-2/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2 - Part 1

- Student: Shabana Patel

## Requirements

Propose two possible datasets you would like to work with and indicate which is your first and which is your second choice.

Upload and shown the .head() of each of your data sets. For each of the proposed datasets, include the following information in text cells:

1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person? A business? An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

## Imports

In [ ]:
import pandas as pd

In [ ]:
# mount google drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
pd.set_option('display.max_columns', None)

## First choice: Heart Failure Prediction Dataset

Load the data

In [ ]:
fpath1 = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week08/Data/heart.csv'

df1 = pd.read_csv(fpath1)

Inspect the data

In [ ]:
df1.head()

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

In [ ]:
df1.shape
print(f'There are {df1.shape[0]} rows and {df1.shape[1]} columns.')
print(f'The rows represent {df1.shape[0]} observations, and the columns represent {df1.shape[1]-1} features and 1 target variable.')

There are 918 rows and 12 columns.
The rows represent 918 observations, and the columns represent 11 features and 1 target variable.


1. Source of the data

> https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction

2. Brief description of data

> This dataset contains 918 rows and 12 columns. It was created by combining five heart datasets over 11 common features.

3. What is the target?

> `HeartDisease` with values
> - 1 (Heart Disease)
> - 0 (Normal)

4. What does one row represent?

> Each row represents one person or patient.



5. Is this a classification or regression problem?

> Classification

6. How many features does the data have?

> The dataset has 11 features.

7. How many rows are in the dataset?

> There are 918 rows in the dataset.


8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

> I don't foresee any challenges at this time.

## Second choice: Fetal Health Classification

Load the data

In [ ]:
fpath2 = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week08/Data/fetal_health.csv'

df2 = pd.read_csv(fpath2)

Inspect the data

In [ ]:
df2.head()

baseline value  accelerations  fetal_movement  uterine_contractions  \
0           120.0          0.000             0.0                 0.000   
1           132.0          0.006             0.0                 0.006   
2           133.0          0.003             0.0                 0.008   
3           134.0          0.003             0.0                 0.008   
4           132.0          0.007             0.0                 0.008   

   light_decelerations  severe_decelerations  prolongued_decelerations  \
0                0.000                   0.0                       0.0   
1                0.003                   0.0                       0.0   
2                0.003                   0.0                       0.0   
3                0.003                   0.0                       0.0   
4                0.000                   0.0                       0.0   

   abnormal_short_term_variability  mean_value_of_short_term_variability  \
0                             73.0                                   0.5   
1                             17.0                                   2.1   
2                             16.0                                   2.1   
3                             16.0                                   2.4   
4                             16.0                                   2.4   

   percentage_of_time_with_abnormal_long_term_variability  \
0                                               43.0        
1                                                0.0        
2                                                0.0        
3                                                0.0        
4                                                0.0        

   mean_value_of_long_term_variability  histogram_width  histogram_min  \
0                                  2.4             64.0           62.0   
1                                 10.4            130.0           68.0   
2                                 13.4            130.0           68.0   
3                                 23.0            117.0           53.0   
4                                 19.9            117.0           53.0   

   histogram_max  histogram_number_of_peaks  histogram_number_of_zeroes  \
0          126.0                        2.0                         0.0   
1          198.0                        6.0                         1.0   
2          198.0                        5.0                         1.0   
3          170.0                       11.0                         0.0   
4          170.0                        9.0                         0.0   

   histogram_mode  histogram_mean  histogram_median  histogram_variance  \
0           120.0           137.0             121.0                73.0   
1           141.0           136.0             140.0                12.0   
2           141.0           135.0             138.0                13.0   
3           137.0           134.0             137.0                13.0   
4           137.0           136.0             138.0                11.0   

   histogram_tendency  fetal_health  
0                 1.0           2.0  
1                 0.0           1.0  
2                 0.0           1.0  
3                 1.0           1.0  
4                 1.0           1.0

In [ ]:
df2.shape
print(f'There are {df2.shape[0]} rows and {df2.shape[1]} columns.')
print(f'The rows represent {df2.shape[0]} observations, and the columns represent {df2.shape[1]-1} features and 1 target variable.')

There are 2126 rows and 22 columns.
The rows represent 2126 observations, and the columns represent 21 features and 1 target variable.


1. Source of the data

> https://www.kaggle.com/datasets/andrewmvd/fetal-health-classification

2. Brief description of data

> This dataset contains 2126 rows and 22 columns. Data is extracted from cardiotocogram exams, which were then classified by three expert obstetritians into 3 classes:
> - Normal
> - Suspect
> - Pathological

3. What is the target?

> `fetal_health` with values
> - 1 (Normal)
> - 2 (Suspect) and
> - 3 (Pathological)

4. What does one row represent?

> Each row represents the results from one cardiotocogram exam.

5. Is this a classification or regression problem?

> Classification

6. How many features does the data have?

> The dataset has 21 features.

7. How many rows are in the dataset?

> There are 2 126 rows in the dataset.


8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

> I don't foresee any challenges at this time.